### DISCUSSION TITLE
Suggesting Ambient Light Feature for HyperHDR

### DISCUSSION THREAD

**Discussion by Rajlodh-001** [awawa-dev/HyperHDR#832] [Ideas]
HyperHDR is a powerful  ambient lighting software, One area where HyperHDR could be further improved is by adding support for ambient lighting similar to Philips Hue's ambient light feature. This feature would enhance the experience by extending the on-screen colors and creating a more immersive viewing environment. 

![Screenshot (55)](https://github.com/awawa-dev/HyperHDR/assets/120379552/37e4ef0a-0296-41fc-97f5-117ebfd85197)
![Screenshot (58)](https://github.com/awawa-dev/HyperHDR/assets/120379552/c73d9128-682a-4d48-bf60-56153f58259e)

Screenshot sample taken From this video :
https://www.youtube.com/watch?v=5MNgigfFpB0

Feature to add in HyperHDR
Ambilight Zones: Divide the screen into zones to create more accurate and precise ambient lighting effects for addition lighting Zones.

for Example :-

![Screenshot (60)](https://github.com/awawa-dev/HyperHDR/assets/120379552/163bd318-46f9-49aa-abbf-73b4369cae42)

Edit on image is done by me for suggestion. 

Screenshot sample taken From this video :
The ULTIMATE Guide to Building an Ambilight TV with Hyperion
https://www.youtube.com/watch?v=J26oYlKyq7Q&t=370s

Option for adding extra light zone for different placement.
The Feature can be set like this :-

Light Zone 1 from led No : 31 - 55 on data pinout GPIO pin 27 on pi
Light Zone 2 from led No : 92 - 116 on data pinout GPIO poi 22 on pi

Multiple Zones as UserWish.

I Would like to suggest this  new feature for Ambilight Zones in HyperHDR ,
And Start a New Discussion .

**Comment by awawa-dev** [comment]
Hi
Currently, in this form on the screenshot (BTW very outdated) you have a context menu after right-clicking, you can add/edit/delete areas used for mapping light sources to the video source. You can use this to tune Philips Hue lamps configuration and define/edit areas that you want for other controllers as well. The second option is grouping feature that allows to combine multiply LEDs into segments that share the same color. However, controlling the Neopixel LED strip with GPIO is not officially supported by HyperHDR (and does not work at all on RPi 5 due to significant changes in hardware including the use of the RP1 bridge) and you have to use other drivers here, e.g. WLED or HyperSerialPico in single or multi-segment mode.
